main.cpp

```diff
@@ -56,64 +56,83 @@ Create a branch named Part1
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    int value;
+    std::string name;
+    
+    T(int v, const char* n) : value(v), name(n) { }   //1, 2, 3
 };
 
-struct <#structName1#>                                //4
+struct StructOne    //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b)    //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if(a != nullptr && b != nullptr)
+        {
+            if(a->value < b->value) return a;
+            if(a->value > b->value) return b;
+        }
+
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float floatOne { 0 }, floatTwo { 0 };
+    float memberFunction(float* updatedValue)    //12 Here and in <staticFunctionA> it seems like I would need to return a nullptr if <updatedValue> turns out to be a nullptr? Otherwise as the consloe puts it I might "reach the end of non-void function"?
     {
-        
+        if(updatedValue != nullptr)
+        {
+            std::cout << "U's floatOne value: " << floatOne << std::endl;
+            floatOne = *updatedValue;
+            std::cout << "U's floatOne updated value: " << floatOne << std::endl;
+            while(std::abs(floatTwo - floatOne) > 0.001f)
+            {
+                floatTwo += 0.001f;
+            }   
+            std::cout << "U's floatTwo updated value: " << floatTwo << std::endl;
+            return floatTwo * floatOne;
+        }
     }
 };
 
-struct <#structname2#>
+struct StructTwo
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float staticFunctionA(U* that, float* updatedValue)    //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if(updatedValue != nullptr)
         {
-            /*
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
-             */
-            that-><#name2#> += ;
+            std::cout << "U's floatOne value: " << that->floatOne << std::endl;
+            that->floatOne = *updatedValue;
+            std::cout << "U's floatOne updated value: " << that->floatOne << std::endl;
+            while( std::abs(that->floatTwo - that->floatOne) > 0.001f)
+            {
+                that->floatTwo += 0.001f;
+            }
+            std::cout << "U's floatTwo updated value: " << that->floatTwo << std::endl;
+            return that->floatTwo * that->floatOne;
         }
```

Comment:
- [ ] add missing return statement

Reply:
added 
`return 0;`
the function doesn't have a pointer as return value so that's the best I could do, right?

Reply:
sorry I missed your previous comment!